src/EditorFeatures/TestUtilities/CodeActions/CSharpCodeFixVerifier`2+Test.cs

```diff
@@ -0,0 +1,88 @@
+            /// </summary>
+            private static readonly ImmutableDictionary<string, ReportDiagnostic> s_nullableWarnings = GetNullableWarningsFromCompiler();
+
+            public Test()
+            {
+                MarkupOptions = Testing.MarkupOptions.UseFirstDescriptor;
+
+                SolutionTransforms.Add((solution, projectId) =>
+                {
+                    var parseOptions = (CSharpParseOptions)solution.GetProject(projectId).ParseOptions;
+                    solution = solution.WithProjectParseOptions(projectId, parseOptions.WithLanguageVersion(LanguageVersion));
+
+                    var compilationOptions = solution.GetProject(projectId).CompilationOptions;
+                    compilationOptions = compilationOptions.WithSpecificDiagnosticOptions(compilationOptions.SpecificDiagnosticOptions.SetItems(s_nullableWarnings));
+                    solution = solution.WithProjectCompilationOptions(projectId, compilationOptions);
+
+                    var options = solution.Options;
+                    foreach (var (key, value) in Options)
+                    {
+                        options = options.WithChangedOption(key, value);
+                    }
+
+                    solution = solution.WithOptions(options);
+
+                    return solution;
+                });
+            }
+
+            private static ImmutableDictionary<string, ReportDiagnostic> GetNullableWarningsFromCompiler()
+            {
+                string[] args = { "/warnaserror:nullable" };
+                var commandLineArguments = CSharpCommandLineParser.Default.Parse(args, baseDirectory: Environment.CurrentDirectory, sdkDirectory: Environment.CurrentDirectory);
+                var nullableWarnings = commandLineArguments.CompilationOptions.SpecificDiagnosticOptions;
+
+                // Workaround for https://github.com/dotnet/roslyn/issues/41610
+                nullableWarnings = nullableWarnings
+                    .SetItem("CS8632", ReportDiagnostic.Error)
+                    .SetItem("CS8669", ReportDiagnostic.Error);
+
+                return nullableWarnings;
+            }
+
+            /// <summary>
+            /// Gets or sets the language version to use for the test. The default value is
+            /// <see cref="LanguageVersion.CSharp8"/>.
+            /// </summary>
+            public LanguageVersion LanguageVersion { get; set; } = LanguageVersion.CSharp8;
+
+            /// <summary>
+            /// Gets a collection of options to apply to <see cref="Solution.Options"/> for testing. Values may be added
+            /// using a collection initializer.
+            /// </summary>
+            public OptionsCollection Options { get; } = new OptionsCollection(LanguageNames.CSharp);
+
+            protected override AnalyzerOptions GetAnalyzerOptions(Project project)
+            {
+                return new WorkspaceAnalyzerOptions(base.GetAnalyzerOptions(project), project.Solution);
```

Comment:
FYI: `WorkspaceAnalyzerOptions` type is not available in CodeStyle layer.

Reply:
WorkspaceAnalyzerOptions is _required_ for the current analyzer implementations. Translation to CodeStyle layer equivalents would be left for the porting process.